cvmfs/server/cvmfs_server_snapshot.sh

```diff
@@ -234,13 +234,36 @@ EOF
     exit 1
   fi
 
+  local maxparallel="${CVMFS_MAX_PARALLEL_SNAPSHOTS:-`nproc`}"
+
+  # make locks in a tmpfs directory so they will go away after system crash
+  local tmpdir=/dev/shm/cvmfs_snapshot_all
+  if [ ! -d $tmpdir ]; then
+    mkdir -m 1777 $tmpdir
+  fi
+  local locknum=0
+  local lockfile=""
+  while [ "$locknum" -lt "$maxparallel" ]; do
+    lockfile=$tmpdir/$locknum
+    if acquire_lock $lockfile; then
+      break
+    fi
+    let locknum+=1
+  done
+  if [ "$locknum" -ge "$maxparallel" ]; then
+    # reached the maximum, silently exit
```

Comment:
I think it would be useful to log something in this case.  We could report in `$fulllog` that the command was skipped.

Reply:
Done, please re-review